core/utils/utils.go

```diff
@@ -950,7 +950,8 @@ func ExtractRevertReasonFromRPCError(err error) (string, error) {
 	if err == nil {
 		return "", errors.New("no error present")
 	}
-	field := reflect.ValueOf(err).Elem().FieldByName("Data")
+	cause := errors.Cause(err)
```

Comment:
Can we move this to `eth/errors.go` or `eth/rpc_errors.go` or similar?

I'd like to keep all the eth error detection in one place if possible.

Reply:
yup!